### PR TITLE
Add dependabot for dependency upgrades

### PR DESCRIPTION
**Pull request by clintonsteiner**
- #341

### CODE REVIEW

**Comment by akaihola** [comment]
Thanks @clintonsteiner!

So does `dependabot` work fine with packages which use `setuptools` and `setup.cfg`? I remember participating in the discussion of dependabot/dependabot-core#2133, but it might be that the issue concerns only the depenedency graph feature on GitHub.

**Review by akaihola — CHANGES_REQUESTED** [review]
Can we somehow test this before merging?

Also, a change log entry would be nice.

**Comment by clintonsteiner** [comment]
https://github.com/clintonsteiner/darker/actions/runs/11605270324/job/32315372582
Testing in fork now appears to work, must enable in code security though
image

**Comment by clintonsteiner** [comment]
https://github.com/akaihola/darker/pull/767
I'd like to get rid of the constraints-oldest and move to pyproject.toml

I was unable to make dependabot acknowledge both files and seems a pain to manage both

File: setup.cfg [hunk 1]
```diff
@@ -60,7 +60,7 @@ color =
     Pygments>=2.4.0
 test =
     # NOTE: remember to keep `constraints-oldest.txt` in sync with these
-    black>=22.3.0
+    black>=24.1.0
```
[the remark this answers is not in the document]
Comment (akaihola) [reply setup.cfg:63]: I do see a [CVE security vulnerability](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-21503) in [Black 24.3.0](/psf/black/blob/main/CHANGES.md#2430) though. But I couldn't find anything before 24.1.0 which would have been fixed in that version.

I guess it makes sense to drop support for Black <24.3.0, but on the other hand I wouldn't like to break users' workflows if they for any reason choose to pin to a selected Black version. And if the code they are formatting is all internal and trusted, I think that's a valid decision.

What do you think?

[the remark this answers is not in the document]
Comment (akaihola) [reply setup.cfg:63]: I wonder how dependabot will react to dependencies in extras – Black will be moved there in #759.

**Comment by akaihola** [comment]
> #767
> I'd like to get rid of the constraints-oldest and move to pyproject.toml
> 
> I was unable to make dependabot acknowledge both files and seems a pain to manage both

I noticed you closed #767 – did you find a work-around?

I wanted to move away from `setup.cfg` as well, and [uv] nowadays provides `--resolution=lowest`, equivalent to what I've used `constraints-oldest.txt` for.

**But:** Due to missing raw HTML support on [PyPI], we still depend on `setup.py` to strip the contributor table from `README.rst` in the distribution packages. This is the only reason we're still using `setup.py` and `setup.cfg`. I've been looking into different ways of solving this (e.g. configuring PyPI or GitHub to use a different file for the project front page), but haven't found an elegant solution yet.

[uv]: https://docs.astral.sh/uv/reference/cli/#:~:text=lowest%20compatible%20version
[PyPI]: https://pypi.org/project/darker

File: setup.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (clintonsteiner) [reply setup.cfg:63]: It tried to update the constraints-oldest.txt file but left the setup.cfg alone

Think we need to have dependencies defined only in one spot for dependabot

[the remark this answers is not in the document]
Comment (akaihola) [reply setup.cfg:63]: Can we instruct dependabot to ignore the Black version? As said, I'd like to not break workflows of anyone choosing to stick to an older Black version e.g. in an in-house trusted code environment.

[the remark this answers is not in the document]
Comment (clintonsteiner) [reply setup.cfg:63]: Added

**Pull request merged**